### ISSUE TITLE
Cannot read 10x bam file

### ISSUE THREAD

**Issue by olgabot** [betteridiot/bamnostic#15]
**Describe the bug**

Cannot read 10x bam file

**To Reproduce**

Using the `possorted_genome_bam.bam` file from [here](https://github.com/dib-lab/sourmash/tree/15af5e08079be2f52d3813c3e006c37d3385ec0e/tests/test-data/lung_ptprc), the following code fails:

```python
import bamnostic as bs

bam_file = bs.AlignmentFile(
    os.path.join(folder, 'possorted_genome_bam.bam'), mode='rb')
```

**Expected behavior**
Should be able to read the file and iterate through alignments

**Screenshots**
```
(sourmash) 
 ⚙  Thu  6 Sep - 13:33  ~/code/sourmash   origin ☊ olgabot/10x-singlecell-bam ↑1 8☀ 3● 3‒ 1✚ ⚑ 
  sourmash compute -k 31 --input-is-10x  --force tests/test-data/lung_ptprc
computing signatures for files: tests/test-data/lung_ptprc
Computing signature for ksizes: [31]
Computing only DNA (and not protein) signatures.
Computing a total of 1 signature(s).
Traceback (most recent call last):
  File "/anaconda3/envs/sourmash/bin/sourmash", line 11, in <module>
    load_entry_point('sourmash', 'console_scripts', 'sourmash')()
  File "/Users/olgabot/code/sourmash/sourmash/__main__.py", line 77, in main
    cmd(sys.argv[2:])
  File "/Users/olgabot/code/sourmash/sourmash/commands.py", line 268, in compute
    barcodes, bam_file = read_10x_folder(filename)
  File "/Users/olgabot/code/sourmash/sourmash/tenx.py", line 17, in read_10x_folder
    os.path.join(folder, 'possorted_genome_bam.bam'), mode='rb')
  File "/Users/olgabot/anaconda/envs/sourmash/lib/python3.6/site-packages/bamnostic/core.py", line 153, in __init__
    bgzf.BgzfReader.__init__(self, **kwargs)
  File "/Users/olgabot/anaconda/envs/sourmash/lib/python3.6/site-packages/bamnostic/bgzf.py", line 471, in __init__
    self._load_header(check_sq)
  File "/Users/olgabot/anaconda/envs/sourmash/lib/python3.6/site-packages/bamnostic/bgzf.py", line 652, in _load_header
    self._header = BAMheader(self)
  File "/Users/olgabot/anaconda/envs/sourmash/lib/python3.6/site-packages/bamnostic/bgzf.py", line 311, in __init__
    tag, value = field.split(':')
ValueError: too many values to unpack (expected 2)
```

**Desktop (please complete the following information):**
 - OS: macOS
 - Python Version: 3.6.5
 - bamnostic Version: 0.8.11

**Additional context**
None

**Comment by olgabot** [comment]
Bamnostic was suggested (https://github.com/dib-lab/sourmash/pull/539) as a replacement to pysam but I'm not able to use it for 10x-generated single cell bam files.

**Comment by betteridiot** [comment]
I saw your issue. I thought by the tag, field part that it might have  been an unpacking issue in the binary part of the data. The problem was actually with how it parses the SAM header. The fields of the SAM header aren't expected to be 3-field because the field type is only really necessary for the BAM format to figure out how to unpack the tag data.

I fixed this to handle 3-field tags in SAM header though. **Thank you** for the heads-up.

I made an update to PyPI and conda. Conda might take awhile until it is finalized. If you want the updates now, feel free to just github install it.
